OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs

```diff
@@ -448,6 +448,75 @@ public async Task GetUniqueChatRoom_WhenRoomDoesNotExist_ShouldReturnNull()
     }
     #endregion
 
+    #region GetCurrentUserUnreadMessagesCountAsync
+    [Test]
+    public async Task GetCurrentUserUnreadMessagesCount_WhenUserRoleIsParent_ShouldReturnCorrectCount()
+    {
+        // Arrange
+        var parentId = parents[0].Id;
+        var role = Role.Parent;
+        var chatRooms = new List<ChatRoomWorkshopForChatList>
+        {
+            new ChatRoomWorkshopForChatList { NotReadByCurrentUserMessagesCount = 1 },
+            new ChatRoomWorkshopForChatList { NotReadByCurrentUserMessagesCount = 1 },
+        };
+
+        var expectedCount = chatRooms.Count(r => r.NotReadByCurrentUserMessagesCount > 0);
+
+        roomWithSpecialModelRepositoryMock
+            .Setup(x => x.GetByParentIdAsync(parentId, It.IsAny<bool>()))
+            .ReturnsAsync(chatRooms);
+
+        // Act
+        var result = await roomService.GetCurrentUserUnreadMessagesCountAsync(parentId, role).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsNotNull(result);
```

Comment:
What does `Assert.IsNotNull(result);` when result is of `int` type? Could `int` be `null`? Why are you checking it?

Reply:
Understood and fixed

Reply:
Thank you